pkg/apis/options/legacy_options.go

```diff
@@ -558,6 +559,7 @@ func legacyProviderFlagSet() *pflag.FlagSet {
 	flagSet.Bool("entra-id-federated-token-auth", false, "enable oAuth client authentication with federated token projected by Azure Workload Identity plugin, instead of client secret.")
 	flagSet.String("bitbucket-team", "", "restrict logins to members of this team")
 	flagSet.String("bitbucket-repository", "", "restrict logins to user with access to this repository")
+	flagSet.StringSlice("discord-restricted-user-id", []string{}, "restrict login to a list of discord user identifiers")
```

Comment:
Should we really add more *legacy* options?
Isn't the setting redundant with the config available via DiscordOptions?

Note: I'm not a maintainer and I'm just discovering the codebase via this PR so my view doesn't count.